src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,291 @@
+                                    /* isAbsolutePathAnUrl */ false
+                                );
+
+                                // Make sure we got back a path that can be a valid module specifier
+                                const isRootedOrRelative = isExternalModuleNameRelative(pathName) || isRootedDiskPath(pathName);
+                                moduleSpecifier = removeFileExtension(isRootedOrRelative ? pathName : combinePaths(".", pathName));
+                            }
+
+                            allActions.push(getCodeActionForImport(moduleSpecifier, (a, b) =>
+                                compareModuleSpecifiers(a, b, sourceDir, useCaseSensitiveFileNames)
+                            ));
+                        }
+                    }
+                }
+            }
+
+            return allActions;
+
+            function convertPathToModuleSpecifier(modulePath: string): string {
+                const i = modulePath.lastIndexOf(nodeModulesDir);
+
+                // This is not a module from the node_module folder, 
+                // so just retun the path
+                if (i === -1) {
+                    return modulePath;
+                }
+
+                // If this is a node module, check to see if the given file is the main export of the module or not. If so,
+                // it can be referenced by just the module name.
+                const moduleDir = getDirectoryPath(modulePath);
+                let nodePackage: any;
+                try {
+                    nodePackage = JSON.parse(context.host.readFile(combinePaths(moduleDir, "package.json")));
+                }
+                catch (e) { }
+
+                // If no main export is explicitly defined, check for the default (index.js)
+                const mainExport = (nodePackage && nodePackage.main) || "index.js";
+                const mainExportPath = normalizePath(isRootedDiskPath(mainExport) ? mainExport : combinePaths(moduleDir, mainExport));
+
+                const moduleSpecifier = removeFileExtension(modulePath.substring(i + nodeModulesDir.length + 1));
+
+                if (compareModuleSpecifiers(modulePath, mainExportPath, getDirectoryPath(sourceFile.fileName), useCaseSensitiveFileNames) === Comparison.EqualTo) {
+                    return getDirectoryPath(moduleSpecifier);
+                }
+                else {
+                    return moduleSpecifier;
+                }
+            }
+
+            function getCodeActionForImport(moduleName: string, isEqual: (a: string, b: string) => Comparison = compareStrings): CodeAction {
+                // Check to see if there are already imports being made from this source in the current file
+                const existing = forEach(imports, (importDeclaration) => {
+                    if (isEqual(removeQuotes(importDeclaration.moduleSpecifier.getText()), moduleName) === Comparison.EqualTo) {

```

Comment:
the fileName is not guaranteed to be the same as the module name. e.g. path mapping. use sourcefile.resolvedModules, build a reverse lookup map and use it here.